src/modules/party/party.controller.test.ts

```diff
@@ -0,0 +1,50 @@
+import { PartyGenerator } from '@ukef-test/support/generator/party-generator';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { when } from 'jest-when';
+
+import { PartyController } from './party.controller';
+import { PartyService } from './party.service';
+
+jest.mock('./party.service');
+
+describe('PartyController', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const partyIdentifier = valueGenerator.stringOfNumericCharacters();
+
+  let partyService: PartyService;
+  let controller: PartyController;
+
+  beforeEach(() => {
+    partyService = new PartyService(null, null);
+    controller = new PartyController(partyService);
+  });
+
+  describe('getPartyByIdentifier', () => {
+    const { parties, partiesFromApi } = new PartyGenerator(valueGenerator).generate({ numberToGenerate: 1 });
+    const partyFromService = parties[0];
+    const expectedParty = partiesFromApi[0];
+
+    it('returns the party from the service', async () => {
+      // eslint-disable-next-line jest/unbound-method
+      when(partyService.getPartyByIdentifier).calledWith(partyIdentifier).mockResolvedValueOnce(partyFromService);
+
+      const party = await controller.getPartyByIdentifier(partyIdentifier);
+
+      expect(party).toStrictEqual(expectedParty);
+    });
+
+    it.skip('does NOT return unexpected keys for the party from the service', async () => {
+      // TODO APIM-78: do we want to do this, or should this be middleware?
+      const partyWithUnexpectedKey = {
+        ...partyFromService,
+        unexpectedKey: valueGenerator.string(),
+      };
+      // eslint-disable-next-line jest/unbound-method
```

Comment:
Inline ESLint please remove.

Reply:
There's not a very good way to satisfy the `unbound-method` rule here:

- there isn't a problem with using an unbound method here, because that method is being mocked
- we can't use an arrow function because that'd be a different function, not the method we want to mock
- we can't annotate with `this: void` because the method _does_ use `this` (when it is not mocked)

Our options for dealing with this are:
1. turn off the rule throughout the project 
2. use inline comments to disable the rule when we know we're mocking the unbound method
3. write our own eslint plugin that extends the rule to know that it's okay to use an unbound method when it's wrapped in a `when` call from `jest-when` (I've not been able to find an existing plugin for this unfortunately)
4. we can replace
    ```javascript
    // eslint-disable-next-line jest/unbound-method
    when(partyService.getPartyByIdentifier).calledWith ...
    ```
    with
    ```javascript
     const partyServiceGetPartyByIdentifier = jest.fn();
     partyService.getPartyByIdentifier = partyServiceGetPartyByIdentifier;
     when(partyServiceGetPartyByIdentifier).calledWith ...
    ```

I don't think we should do 1 (because the rule is usually useful) or 3 (because it seems like a lot of work we don't want to bring into this project).

Out of 2 and 4, I preferred 2 because I think it's easier to read. What do you think @abhi-markan?

Reply:
Hello @lmarrai-sw if we can go with option 4 please that will be great, understand it will cause code repetition but would prefer it over inline rule disable. Happy to go over other suggestions.